blog/src/Auth/Controller/ChangePasswordController.php

```diff
@@ -0,0 +1,95 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Auth\Controller;
+
+use App\Auth\AuthService;
+use App\Auth\Identity;
+use App\Auth\IdentityRepository;
+use App\Auth\Form\ChangePasswordForm;
+use App\Service\WebControllerService;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Yiisoft\FormModel\FormHydrator;
+use Yiisoft\Http\Method;
+use Yiisoft\Session\SessionInterface as Session;
+use Yiisoft\Session\Flash\Flash;
+use Yiisoft\Translator\TranslatorInterface as Translator;
+use Yiisoft\User\CurrentUser;
+use Yiisoft\Yii\View\ViewRenderer;
+
+final class ChangePasswordController
+{
+    public function __construct(
+      private Session $session,
+      private Flash $flash,
+      private Translator $translator,
+      private CurrentUser $current_user,
+      private WebControllerService $webService, 
+      private ViewRenderer $viewRenderer,
+    )
+    {
+      $this->current_user = $current_user;
+      $this->session = $session;      
+      $this->flash = new Flash($session);
+      $this->translator = $translator;
+      $this->viewRenderer = $viewRenderer->withControllerName('changepassword');
+    }
+    
+    public function change(
+      AuthService $authService,
+      Identity $identity,
+      IdentityRepository $identityRepository,
+      ServerRequestInterface $request,
+      FormHydrator $formHydrator,
+      ChangePasswordForm $changePasswordForm
+    ): ResponseInterface {
+      if ($authService->isGuest()) {
+          return $this->redirectToMain();
+      }  
+      // permit an authenticated user, ie. not a guest, only and null!== current user
+      if (!$authService->isGuest()) {
```

Comment:
`if (!$authService->isGuest()) {` already not need, we have this check above.

In the same way, you can remove "if" nesting in further code.